myconnections/src/java/org/sakaiproject/myconnections/ui/WidgetPage.java

```diff
@@ -0,0 +1,175 @@
+	 */
+	int maxUsers = 60;
+	int cols = 4;
+
+	public WidgetPage() {
+		log.debug("WidgetPage()");
+	}
+
+	@Override
+	public void onInitialize() {
+		super.onInitialize();
+
+		// get maxUsers, cols
+		this.maxUsers = this.serverConfigurationService.getInt("widget.myconnections.maxusers", this.maxUsers);
+		this.cols = this.serverConfigurationService.getInt("widget.myconnections.cols", this.cols);
+
+		// get current user
+		final String currentUserUuid = this.sessionManager.getCurrentSessionUserId();
+		
+		// get requests
+		List<? extends BasicPerson> requests = this.connectionsLogic.getConnectionRequestsForUser(currentUserUuid);
+		
+		// sort
+		Collections.sort(requests);
+
+		List<GridPerson> requestGridPersons = sliceAndFill(requests, "request", false);
+
+		// get connections, sort and slice
+		List<? extends BasicConnection> connections = this.connectionsLogic.getBasicConnectionsForUser(currentUserUuid);
+
+		// sort
+		Collections.sort(connections, new Comparator<BasicConnection>() {
+
+			@Override
+			public int compare(final BasicConnection o1, final BasicConnection o2) {
+				return new CompareToBuilder()
+						.append(o1.getOnlineStatus()==ProfileConstants.ONLINE_STATUS_OFFLINE?2:1,
+								o2.getOnlineStatus()==ProfileConstants.ONLINE_STATUS_OFFLINE?2:1)
+						.append(o1.getDisplayName(), o2.getDisplayName())
+						.toComparison();
+			}
+
+		});
+
+		List<GridPerson> connectionGridPersons = sliceAndFill(connections, "connection", true);
+
+		// concatenate
+		final List<GridPerson> finalList = new ArrayList<GridPerson>();
+		finalList.addAll(requestGridPersons);
+		finalList.addAll(connectionGridPersons);
+
+		// add connections grid or label
+		if (!finalList.isEmpty()) {
+			add(new ConnectionsGrid("connections", Model.ofList(finalList), cols));
+		} else {
+			add(new Label("connections", new ResourceModel("label.noconnections"))
+					.add(new AttributeAppender("class", "instruction")));
+		}
+
+	}
+
+	@Override
+	public void renderHead(final IHeaderResponse response) {
+		super.renderHead(response);
+		
+		final String version = this.serverConfigurationService.getString("portal.cdn.version", "");
+
+		// get the Sakai skin header fragment from the request attribute
+		final HttpServletRequest request = (HttpServletRequest) getRequest().getContainerRequest();
+
+		response.render(StringHeaderItem.forString((String) request.getAttribute("sakai.html.head")));
+		response.render(OnLoadHeaderItem.forScript("setMainFrameHeight( window.name )"));
+
+		// Tool additions (at end so we can override if required)
+		response.render(StringHeaderItem.forString("<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />"));
+
+		// render jQuery and the Wicket event library
+		// Both must be priority so they are emitted into the head
+		response.render(new PriorityHeaderItem(JavaScriptHeaderItem.forUrl(String.format("/library/webjars/jquery/1.11.3/jquery.min.js?version=%s", version))));
```

Comment:
Should maybe use org.sakaiproject.portal.util.PortalUtils#getWebjarsPath to get the webjar path.